pkg/antctl/raw/supportbundle/command_test.go

```diff
@@ -0,0 +1,369 @@
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := createControllerClient(ctx, tt.k8sClientset, tt.antreaClientset, clientConfig)
+			if tt.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			}
+		})
+	}
+}
+
+func TestCreateAgentClients(t *testing.T) {
+	tests := []struct {
+		name            string
+		expectedErr     string
+		k8sClientset    *fake.Clientset
+		antreaClientset *fakeclientset.Clientset
+		prepareReactor  func(antreaClientset *fakeclientset.Clientset, k8sClientset *fake.Clientset)
+		expectedClients []string
+	}{
+		{
+			name:            "created both agent clients successfully",
+			k8sClientset:    fake.NewSimpleClientset(&node1, &node3),
+			antreaClientset: fakeclientset.NewSimpleClientset(agentInfo1, agentInfo2),
+			expectedClients: []string{"node-1", "node-3"},
+		},
+		{
+			name:            "failure to create one agent client due to error when parsing controller IP",
+			k8sClientset:    fake.NewSimpleClientset(&node2, &node3),
+			antreaClientset: fakeclientset.NewSimpleClientset(agentInfo1, agentInfo2),
+			expectedClients: []string{"node-3"},
+		},
+		{
+			name:            "no-ops due to error in listing antreaagentinfos",
+			antreaClientset: fakeclientset.NewSimpleClientset(),
+			prepareReactor: func(antreaClientset *fakeclientset.Clientset, k8sClientset *fake.Clientset) {
+				antreaClientset.PrependReactor("list", "antreaagentinfos", func(action cgtesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &v1beta1.AntreaAgentInfoList{}, errors.New("error listing antreaagentinfos")
+				})
+			},
+			expectedErr: "error listing antreaagentinfos",
+		},
+		{
+			name:            "no-ops due to error in listing nodes",
+			antreaClientset: fakeclientset.NewSimpleClientset(),
+			k8sClientset:    fake.NewSimpleClientset(),
+			prepareReactor: func(antreaClientset *fakeclientset.Clientset, k8sClientset *fake.Clientset) {
+				k8sClientset.PrependReactor("list", "nodes", func(action cgtesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &v1.NodeList{}, errors.New("error listing nodes")
+				})
+			},
+			expectedErr: "error listing nodes",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.prepareReactor != nil {
+				tt.prepareReactor(tt.antreaClientset, tt.k8sClientset)
+			}
+			clients, err := createAgentClients(tt.k8sClientset, tt.antreaClientset, clientConfig, "", nameList)
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+				agentClients := []string{}
+				for client := range clients {
+					agentClients = append(agentClients, client)
+				}
+				sort.Strings(agentClients)
+				assert.ElementsMatch(t, tt.expectedClients, agentClients)
+			}
+		})
+	}
+}
+
+func TestRequest(t *testing.T) {
+	agentClients := map[string]*rest.RESTClient{}
```

Comment:
same comment as above, it would make more sense to use `fake.RESTClient` whenever possible.